cuda/kernels.cu

```diff
@@ -82,83 +93,109 @@ __global__ void organize_points(Point *d_points, int *d_categories, Point *bl,
 	__syncthreads();
 
 	int start = threadIdx.x * range;
-	for (int i = start; i < start + range; i++) {
-		if (i < count) {
+	for (int i = start; i < start + range; i++)
+	{
+		if (i < count)
+		{
 			// Point array will store the respective points in a contiguous
 			// fashion increment subgrid index according to the category
 			unsigned int category_index =
 				atomicAdd(&subgrid_index[d_categories[i]], 1);
-			if (d_categories[i] == 0) {
+			if (d_categories[i] == 0)
+			{
 				bl[category_index] = d_points[i];
 			}
-			if (d_categories[i] == 1) {
+			if (d_categories[i] == 1)
+			{
 				br[category_index] = d_points[i];
 			}
-			if (d_categories[i] == 2) {
+			if (d_categories[i] == 2)
+			{
 				tl[category_index] = d_points[i];
 			}
-			if (d_categories[i] == 3) {
+			if (d_categories[i] == 3)
+			{
 				tr[category_index] = d_points[i];
 			}
 		}
 	}
 }
 
+// Quandrant Search to find the level of the quadrant where the point lies
+__global__ void quadrant_search(Point *target_point, QuadrantBoundary *boundaries, int num_boundaries, int *result)
```

Comment:
I think the current kernel is using all threads to access specific indexes withing QuadrantBoundaries, instead we can have a single thread check the entire array of boundaries for a single query (would be much better utilisation that way as we previously discussed). Also I'd suggest that we move QuadrantBoundaries to constant memory since it would be accessed by multiple threads and all positions too, it would be faster than global memory.